File: src/test/javascript/examples/geometry/Point.js
var Vector = require("./Vector");

/**
 * @constructor
 */
function Point(x, y) {
	Vector.apply(this, arguments);
}

require("util").inherits(Point, Vector);

var origin = new Point(0, 0);

/**
 * @return {Point} the origin
 * @static
 */
function Point$getOrigin() {
	return origin;
}

Point.prototype.getOrigin = Point$getOrigin;

module.exports = Point;

File: src/test/java/examples/geometry/Point.java
package examples.geometry;

public class Point extends Vector {
	Point(double... arguments) {
		super(arguments);
	}

	static Point origin = new Point(0, 0);

	/**
	 * @return {Point} the origin
	 */
	public static Point getOrigin() {
		return origin;
	}

}
